### PR TITLE
refactor: remove duplicated ref to player can be accessed from client

### PR DESCRIPTION
**Pull request by uweenukr**
There was multiple ways of getting the same information and while that can be useful it can also be confusing to new users. It also provides opportunities that some values may not be set at all or in the correct order. Do you think the reduction in API redundancy and size is a good thing?

BREAKING

Old: ClientObjectManager.LocalPlayer
New: Client.Player.Identity    (note: this worked before the change too)

### CODE REVIEW

File: Assets/Mirage/Runtime/ClientObjectManager.cs [hunk 1]
```diff
@@ -515,7 +510,7 @@ internal void OnRpcMessage(RpcMessage msg)
 
         void CheckForLocalPlayer(NetworkIdentity identity)
         {
-            if (identity && identity == LocalPlayer)
+            if (identity && identity == Client.Player.Identity)
```
Comment (James-Frowen) [inline Assets/Mirage/Runtime/ClientObjectManager.cs:513]: do we need to keep the null propagation? the previous use was `Client.Player?.Identity`

Comment (uweenukr) [reply Assets/Mirage/Runtime/ClientObjectManager.cs]: fixed

File: Assets/Mirage/Runtime/NetworkIdentity.cs [hunk 2]
```diff
@@ -134,7 +134,7 @@ public sealed class NetworkIdentity : MonoBehaviour
         /// This returns true if this object is the one that represents the player on the local machine.
         /// <para>This is set when the server has spawned an object for this particular client.</para>
         /// </summary>
-        public bool IsLocalPlayer => ClientObjectManager != null && ClientObjectManager.LocalPlayer == this;
+        public bool IsLocalPlayer => Client != null && Client.Player.Identity == this;
```
Comment (James-Frowen) [inline Assets/Mirage/Runtime/NetworkIdentity.cs:137]:
```suggestion
        public bool IsLocalPlayer => Client != null && Client.Player?.Identity == this;
```

I think this one needs to be nullable as well

Comment (uweenukr) [reply Assets/Mirage/Runtime/NetworkIdentity.cs]: good catch

**Review by James-Frowen — APPROVED** [review]

**Pull request merged**